tracer/src/Datadog.Trace/Span.cs

```diff
@@ -325,8 +326,7 @@ internal string GetTag(string key)
             switch (key)
             {
                 case Trace.Tags.SamplingPriority:
-                    var samplingPriority = Context.TraceContext?.SamplingPriority ?? Context.SamplingPriority;
```

Comment:
I think it is safe to say that `Context.TraceContext` can never be null because we ensure span objects will always have a non-null `Context.TraceContext` when we create them. However, `Context.TraceContext?.SamplingPriority` can still be null if span is a root span (e.g. web server) and the incoming sampling priority was null, as unusual as that is.

We might need to return `Context.TraceContext?.SamplingPriority?.ToString()`